tools/worker-runner/registration/registration_test.go

```diff
@@ -112,6 +115,8 @@ func TestCredsExpirationNewCredentials(t *testing.T) {
 
 	fmt.Printf("setting fake expires to %s\n", expires)
 	reg := new(&runnercfg, &state, tc.FakeWorkerManagerClientFactory)
+	reg.credsExpireCond = sync.NewCond(&sync.Mutex{})
+	reg.credsExpireCond.L.Lock()
```

Comment:
do I read this correctly that this lock will happen after the lock in `reg.credsExpireTimer`?

Reply:
No, this one should happen first.  The timer isn't started until `reg.WorkerStarted`.

The lock is basically held for the duration of the test, except that the test itself releases the lock implicitly by calling `Wait`, and the Timer callback locks, notifies, and unlocks it during that time.